### PR TITLE
Remove WEB3_INFURA_API_KEY environment variable in favor of WEB3_INFURA_PROJECT_ID

### PR DESCRIPTION
**Pull request by kclowes**
### What was wrong?

Another v6 breaking change. We had 2 environment variables that were aliases of one another for backwards compatibility.

### How was it fixed?
Standardized on WEB3_INFURA_PROJECT_ID to match up with Infura's wording.

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: web3/auto/infura/endpoints.py [hunk 1]
```diff
@@ -27,10 +27,7 @@
 
 
 def load_api_key() -> str:
```
Comment (fselmo) [inline web3/auto/infura/endpoints.py:29]: nit: `load_api_key()` -> `load_project_id()`? I'm good with either tbh.

**Review by fselmo — APPROVED** [review]
lgtm 👍🏼... just a tiny nit; take or leave

**Review by pacrob — COMMENTED** [review]
https://github.com/ethereum/web3.py/blob/845bce4921411ecec93713cbf7b29fb82c03182c/web3/exceptions.py#L221

There's still an exception `InfuraKeyNotFound` - should that change to `InfuraIdNotFound` or similar?

**Comment by kclowes** [comment]
Good call, @pacrob. Will update that too!

**Review by pacrob — APPROVED** [review]
ship it!

**Pull request merged**